Microsoft.Toolkit.Uwp.UI.Controls/Graph/SharePointFiles/SharePointFiles.cs

```diff
@@ -0,0 +1,558 @@
+                        _delete.Visibility = Visibility.Collapsed;
+                        Task<IDriveItemPermissionsCollectionPage> taskPermissions = _graphClient.Drives[_driveId].Items[driveItem.Id].Permissions.Request().GetAsync(_cancelGetDetails.Token);
+                        IDriveItemPermissionsCollectionPage permissions = await taskPermissions;
+                        if (!taskPermissions.IsCanceled)
+                        {
+                            foreach (Permission permission in permissions)
+                            {
+                                if (permission.Roles.Contains("write") || permission.Roles.Contains("owner"))
+                                {
+                                    _share.Visibility = Visibility.Visible;
+                                    _delete.Visibility = Visibility.Visible;
+                                    break;
+                                }
+                            }
+
+                            IsDetailPaneVisible = true;
+                            Task<IDriveItemThumbnailsCollectionPage> taskThumbnails = _graphClient.Drives[_driveId].Items[driveItem.Id].Thumbnails.Request().GetAsync(_cancelGetDetails.Token);
+                            IDriveItemThumbnailsCollectionPage thumbnails = await taskThumbnails;
+                            if (!taskThumbnails.IsCanceled)
+                            {
+                                ThumbnailSet thumbnailsSet = thumbnails.FirstOrDefault();
+                                if (thumbnailsSet != null)
+                                {
+                                    _thumbnailImage.Source = new BitmapImage(new Uri(thumbnailsSet.Large.Url));
+                                }
+                            }
+                        }
+                    }
+                    catch
+                    {
+                    }
+                }
+                else
+                {
+                    SelectedFile = null;
+                    FileSize = 0;
+                    LastModified = string.Empty;
+                    _download.Visibility = Visibility.Collapsed;
+                    _share.Visibility = Visibility.Collapsed;
+                    _delete.Visibility = Visibility.Collapsed;
+                    IsDetailPaneVisible = false;
+                }
+            }
+        }
+
+        private void HideDetailsPane()
+        {
+            _list.Visibility = Visibility.Visible;
+            _list.SetValue(Grid.RowSpanProperty, 3);
+            _list.SetValue(Grid.ColumnSpanProperty, 3);
+            _thumbnail.Visibility = Visibility.Collapsed;
+            _details.Visibility = Visibility.Collapsed;
+            if (_driveItemPath.Count <= 1)
+            {
+                _back.Visibility = Visibility.Collapsed;
+            }
+        }
+
+        private void ShowDetailsPane()
+        {
+            _thumbnail.Visibility = Visibility.Visible;
```

Comment:
As mentioned above, these two methods should all be part of the XAML use properties and/or visual states.